pkg/cri/server/sandbox_run.go

```diff
@@ -281,6 +267,39 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		return nil, fmt.Errorf("failed to get sandbox container info: %w", err)
 	}
 
+	if podNetwork {
```

Comment:
i'm still thinking the podNetwork setup should be done in parallel with a wg at the end for moving state to ready if both the pause container/task are started and the network is setup without error..

Reply:
Hmm, that is an interesting idea to explore but we may not want to scope it into this PR.

Reply:
ok will take this idea to another issue..